racetrack_module.py

```diff
@@ -0,0 +1,588 @@
+            point_end = top_right
+        else:
+            vec = horizontal_vec
+            vec_normalized = vec.normalized() * -1
+            point_start = top_left
+            point_end = bottom_left
+
+        unit_factor = QgsUnitTypes.fromUnitToUnitFactor(
+            QgsUnitTypes.DistanceUnit.DistanceMeters,
+            coverage_crs.mapUnits(),
+        )
+
+        return {
+            'vec': vec,
+            'vec_normalized': vec_normalized,
+            'point_start': point_start,
+            'point_end': point_end,
+            'coverage_range': self.coverage_module.compute_sensor_coverage_in_meters(
+                float(self.settings.value("science_flight_planner/sensors", {})[self.sensor_combobox.currentText()]),
+                self.flight_altitude_spinbox.value()
+            ) * unit_factor,
+            'overlap_factor': 1 - float(self.settings.value("science_flight_planner/overlap", 0)),
+            'max_turn_distance': float(self.settings.value("science_flight_planner/max_turn_distance", 1000))
+        }
+
+    def _handle_remaining_tracks(self, j: int, number_of_lines: int,
+                                 max_flyover: int, inner_iteration: int,
+                                 left_point: bool, params: dict) -> List[QgsPointXY]:
+        """Handle the remaining tracks for back and forth algorithm"""
+        points = []
+        forward = True
+        remaining_tracks = (
+            number_of_lines - j if inner_iteration == 0
+            else int((((2 * max_flyover) - inner_iteration) - 2) / 2)
+        )
+
+        for i in range(remaining_tracks):
+            if forward:
+                j = j + remaining_tracks
+            else:
+                j = j - remaining_tracks
+
+            points.extend(self._compute_line_points(
+                j, left_point, params['point_start'], params['point_end'],
+                params['vec_normalized'], params['coverage_range'],
+                params['overlap_factor']
+            ))
+            left_point = not left_point
+            remaining_tracks = remaining_tracks - 1
+            forward = not forward
+            left_point = not left_point
+
+        return points
+
+    def _update_position(self, j: int, forward: bool, max_flyover: int,
+                         line_from_bottom: int, number_of_lines: int) -> int:
+        """Update position for fly to top algorithm"""
+        if forward:
+            if j + max_flyover > number_of_lines:
+                if j + 1 <= number_of_lines:
+                    return j + 1
+                else:
+                    return j + 1 - max_flyover
+            else:
+                return j + max_flyover
+        else:
+            if j == line_from_bottom:
+                return j + 1
+            elif j - max_flyover < line_from_bottom:
+                return line_from_bottom
+            else:
+                return j - max_flyover
+
+    def _get_save_file_path(self, base_path: str,
```

Comment:
Method can be made static
```suggestion
    @staticmethod
    def _get_save_file_path(base_path: str,
```

Reply:
this is a helper-function, only used in the class, so i would prefer keeping it as class method with a leading underscore